pkg/tnf/handlers/nodeport/nodeport.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package nodeport
+
+import (
+	"strings"
+	"time"
+
+	"github.com/test-network-function/test-network-function/pkg/tnf"
+	"github.com/test-network-function/test-network-function/pkg/tnf/identifier"
+	"github.com/test-network-function/test-network-function/pkg/tnf/reel"
+)
+
+const (
+	npRegex = "(?s).+"
+)
+
+type NodePort struct {
+	result  int
+	timeout time.Duration
+	args    []string
+}
+
+func NewNodePort(timeout time.Duration, podNamespace string) *NodePort {
+	return &NodePort{
+		timeout: timeout,
+		result:  tnf.ERROR,
+		args: []string{"oc -n " +
+			podNamespace +
+			" get services -o custom-columns=TYPE:.spec.type | grep -E 'NodePort|TYPE'"},
+	}
+}
+
+// Args returns the command line args for the test.
+func (np *NodePort) Args() []string {
+	return np.args
+}
+
+// GetIdentifier returns the tnf.Test specific identifier.
+func (np *NodePort) GetIdentifier() identifier.Identifier {
+	return identifier.NodePortIdentifier
+}
+
+// Timeout returns the timeout in seconds for the test.
+func (np *NodePort) Timeout() time.Duration {
+	return np.timeout
+}
+
+// Result returns the test result.
+func (np *NodePort) Result() int {
+	return np.result
+}
+
+// ReelFirst returns a step which expects the ping statistics within the test timeout.
+func (np *NodePort) ReelFirst() *reel.Step {
+	return &reel.Step{
+		Expect:  []string{npRegex},
+		Timeout: np.timeout,
+	}
+}
+
+func (np *NodePort) ReelMatch(_, _, match string) *reel.Step {
+	numExpectedLines := 1
```

Comment:
I assume that the one expected line is just the header:
```
NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
```
Might be helpful to add a comment if that is the case, in case someone has to debug this later :).